internals/secrethub/generate.go

```diff
@@ -60,17 +62,24 @@ func NewGenerateSecretCommand(io ui.IO, newClient newClientFunc) *GenerateSecret
-	clause.Flag("min", "<charset>:<n> Ensure that the resulting password contains at least n characters from the given character set. Note that adding constraints reduces the strength of the secret. When possible, avoid any constraints.").SetValue(&cmd.mins)
-	clause.Flag("clip", "Copy the generated value to the clipboard. The clipboard is automatically cleared after "+units.HumanDuration(cmd.clearClipboardAfter)+".").Short('c').BoolVar(&cmd.copyToClipboard)
-	clause.Flag("charset", "Define the set of characters to randomly generate a password from. Options are all, alphanumeric, numeric, lowercase, uppercase, letters, symbols and human-readable. Multiple character sets can be combined by supplying them in a comma separated list. Defaults to alphanumeric.").Default("alphanumeric").HintOptions("all", "alphanumeric", "numeric", "lowercase", "uppercase", "letters", "symbols", "human-readable").SetValue(&cmd.charsetFlag)
-	clause.Flag("symbols", "Include symbols in secret.").Short('s').Hidden().SetValue(&cmd.symbolsFlag)
-	clause.Arg("rand-command", "").Hidden().StringVar(&cmd.secondArg)
-	clause.Arg("length", "").Hidden().SetValue(&cmd.lengthArg)
-
-	command.BindAction(clause, cmd.Run)
+	clause := r.CreateCommand("generate", "Generate a random secret.")
+	clause.Args = cobra.RangeArgs(1, 3)
+	//clause.Arg("secret-path", "The path to write the generated secret to").Required().PlaceHolder(secretPathPlaceHolder).StringVar(&cmd.firstArg)
+	clause.VarP(&cmd.lengthFlag, "length", "l", "The length of the generated secret. Defaults to "+strconv.Itoa(defaultLength), true, false) //.PlaceHolder(strconv.Itoa(defaultLength)).Short('l').SetValue(&cmd.lengthFlag)
+	clause.Flag("length").DefValue = strconv.Itoa(defaultLength)
+	clause.Var(&cmd.mins, "min", "<charset>:<n> Ensure that the resulting password contains at least n characters from the given character set. Note that adding constraints reduces the strength of the secret. When possible, avoid any constraints.", true, false)
+	clause.BoolVarP(&cmd.copyToClipboard, "clip", "c", false, "Copy the generated value to the clipboard. The clipboard is automatically cleared after "+units.HumanDuration(cmd.clearClipboardAfter)+".", true, false)
+	clause.Var(&cmd.charsetFlag, "charset", "Define the set of characters to randomly generate a password from. Options are all, alphanumeric, numeric, lowercase, uppercase, letters, symbols and human-readable. Multiple character sets can be combined by supplying them in a comma separated list. Defaults to alphanumeric.", true, false) //Default("alphanumeric").HintOptions("all", "alphanumeric", "numeric", "lowercase", "uppercase", "letters", "symbols", "human-readable")
+	clause.Flag("charset").DefValue = "alphanumeric"
+	_ = clause.RegisterFlagCompletionFunc("charset", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

Comment:
This error should be handled or returned.

Reply:
This is done in the `Register` method, which does not return any errors. If it were to return the error from line 74, we would need to change the signature of the method, which is not a good aproach since here is the only place where an error may be returned. Also, the error is blank here since the linter made me ignore it explicitly. The flag completions function can also be dynamic, therefore the call's signature returns an error. In our case, it is static, which means that the chances of having an error are close to 0.
I believe we should leave it as it is, since it is the only exception and handling it complicates the infrastructure. Let me know if you have another perspective

Reply:
It makes sense to me. Thanks for clarifying